pkg/output/azureeventhub/azure_event_hub.go

```diff
@@ -0,0 +1,80 @@
+// Licensed to Elasticsearch B.V. under one or more agreements.
+// Elasticsearch B.V. licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+package azureeventhub
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/stream/pkg/output"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs"
+)
+
+func init() {
+	output.Register("azureeventhub", New)
+}
+
+type Output struct {
+	opts           *output.Options
+	producerClient *azeventhubs.ProducerClient
+	cancelFunc     context.CancelFunc
+	cancelCtx      context.Context
+}
+
+func New(opts *output.Options) (output.Output, error) {
+	var producerClient *azeventhubs.ProducerClient
+	var err error
+
+	if opts.AzureEventHubOptions.ConnectionString != "" {
+		producerClient, err = azeventhubs.NewProducerClientFromConnectionString(opts.AzureEventHubOptions.ConnectionString, opts.AzureEventHubOptions.EventHubName, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error while creating new eventhub producer client from connectionstring : %w", err)
+		}
+	} else {
+		fmt.Print("no connectionstring was provided, falling back to default credentials or environment variable")
+
+		// Credentials set as env variables - https://github.com/Azure/azure-sdk-for-go/blob/6b6f76ebe0d2334c83e8b6f89af4fe9d0b1ce631/sdk/azidentity/README.md?plain=1#L156-L187
+		defaultAzureCred, err := azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			return nil, fmt.Errorf("missing azure credentials in the environment variables : %w", err)
+		}
+
+		producerClient, err = azeventhubs.NewProducerClient(opts.AzureEventHubOptions.FullyQualifiedNamespace, opts.AzureEventHubOptions.EventHubName, defaultAzureCred, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error while creating new eventhub producer client : %w", err)
+		}
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	return &Output{opts: opts, producerClient: producerClient, cancelFunc: cancel, cancelCtx: ctx}, nil
+}
+
+func (*Output) DialContext(_ context.Context) error {
+	return nil
+}
+
+func (o *Output) Close() error {
+	o.producerClient.Close(o.cancelCtx)
```

Comment:
The really do this? That's awful; they don't even use the context in the close path.

Reply:
Ofcourse... Do you have any alternative suggestions against this?

Reply:
No, it we have to use it as it is provided.